osquery/config/config.cpp

```diff
@@ -61,278 +111,161 @@ Status Config::update(const std::map<std::string, std::string>& config) {
-  std::string key = option.first.data();
-  std::string value = option.second.data();
-
-  Flag::updateValue(key, value);
-  // There is a special case for supported Gflags-reserved switches.
-  if (key == "verbose" || key == "verbose_debug" || key == "debug") {
-    setVerboseLevel();
-    if (Flag::getValue("verbose") == "true") {
-      VLOG(1) << "Verbose logging enabled by config option";
-    }
-  }
-
-  conf.options[key] = value;
-  if (conf.all_data.count("options") > 0) {
-    conf.all_data.get_child("options").erase(key);
-  }
-  conf.all_data.add_child("options." + key, option.second);
-}
-
-static void additionalScheduledQuery(const std::string& name,
-                                     const tree_node& node,
-                                     ConfigData& conf) {
-  // Read tree/JSON into a query structure.
-  ScheduledQuery query;
-  query.query = node.second.get<std::string>("query", "");
-  query.interval = node.second.get<int>("interval", 0);
-  if (query.interval == 0) {
-    VLOG(1) << "Setting invalid interval=0 to 84600 for query: " << name;
-    query.interval = 86400;
-  }
-
-  // This is a candidate for a catch-all iterator with a catch for boolean type.
-  query.options["snapshot"] = node.second.get<bool>("snapshot", false);
-  query.options["removed"] = node.second.get<bool>("removed", true);
-
-  // Check if this query exists, if so, check if it was changed.
-  if (conf.schedule.count(name) > 0) {
-    if (query == conf.schedule.at(name)) {
-      return;
-    }
-  }
-
-  // This is a new or updated scheduled query, update the splay.
-  query.splayed_interval =
-      splayValue(query.interval, FLAGS_schedule_splay_percent);
-  // Update the schedule map and replace the all_data node record.
-  conf.schedule[name] = query;
-}
-
-inline void mergeScheduledQuery(const std::string& name,
-                                const tree_node& node,
-                                ConfigData& conf) {
-  // Add the new query to the configuration.
-  additionalScheduledQuery(name, node, conf);
-  // Replace the all_data node record.
-  if (conf.all_data.count("schedule") > 0) {
-    conf.all_data.get_child("schedule").erase(name);
-  }
-  conf.all_data.add_child("schedule." + name, node.second);
-}
-
-inline void mergeExtraKey(const std::string& name,
-                          const tree_node& node,
-                          ConfigData& conf) {
-  // Automatically merge extra list/dict top level keys.
-  for (const auto& subitem : node.second) {
-    if (node.second.count("") == 0 && conf.all_data.count(name) > 0) {
-      conf.all_data.get_child(name).erase(subitem.first);
+    if (tree.count("scheduledQueries") > 0) {
+      auto& scheduled_queries = tree.get_child("scheduledQueries");
+      pt::ptree queries;
+      for (const std::pair<std::string, pt::ptree>& query : scheduled_queries) {
+        std::string query_name;
+        try {
+          query_name = query.second.get<std::string>("name");
```

Comment:
You can remove this `try`, `catch` with: `query_name = query.second.get("name", "");`

And then:
```cpp
if (query_name.size() == 0) {
  return Status(1, "....");
}
```